apps/src/studio/blocks.js

```diff
@@ -965,14 +943,22 @@ exports.install = function(blockly, blockInstallOptions) {
       dropdown.setValue(this.VALUES[1][1]); // default to top-left
       this.setHSV(184, 1.0, 0.74);
       if (spriteCount > 1) {
-        this.appendDummyInput().appendTitle(
-          spriteNumberTextDropdown(msg.setSpriteN),
-          'SPRITE'
+        this.interpolateMsg(
+          msg.setSpritePosition(),
+          () => {
+            this.appendValueInput('SPRITE').appendTitle({spriteIndex: ''});
```

Comment:
it's not at all clear to me how this version of the block matches the original. Where is `spriteIndex` coming from? Why do we no longer care about the functionality provided by `spriteNumberTextDropdown`?